python/requirements-dev.txt

```diff
@@ -2,5 +2,7 @@ black==23.3.0
 mypy==1.3.0
 pandas-stubs==2.0.2.230605
 pytest==7.3.1
+pytest-cov==4.1.0
```

Comment:
These deps were missing before because tests weren't running in CI, so we add them here